### DISCUSSION TITLE
How to update collection field name from code?

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2443] [Q&A]
How to update collection field name from code? Should I delete field and add it again?

**Comment by ganigeorgiev** [comment]
Removing the field and creating a new one will delete all existing data associated to that field (unless you preserve the field id).

If you just want to rename the field, you can use the sample from https://github.com/pocketbase/pocketbase/discussions/2432#discussioncomment-5823827. For example:

```go
dao := daos.New(db)

collection, err := dao.FindCollectionByNameOrId("branches")
if err != nil {
    return err
}

// this returns a pointer so we can directly modify the underlying field data
field := collection.Schema.GetFieldByName("oldFieldName")
field.Name = "yourNewFieldName"

dao.SaveCollection(collection)
```
